paywall/src/utils/locks.js

```diff
@@ -15,32 +15,77 @@ export function currencySymbolForLock(lock, config) {
   return currency
 }
 
+/**
+ * Returns true if the keys for `lock` are priced in ERC20, false otherwise.
+ * @param {*} lock
+ */
+export const isERC20Lock = lock => {
+  return !!lock.currencyContractAddress
+}
+
+/**
+ * Returns true if the keys for `lock` are priced in Eth, false otherwise.
+ * @param {*} lock
+ */
+export const isEthLock = lock => {
+  return !lock.currencyContractAddress
+}
+
+// Just setting up named constants here so the code is more easily read, these
+// shouldn't be used outside of this file.
+const LOCK_IS_TOO_EXPENSIVE = true
+const LOCK_IS_AFFORDABLE = false
+
+/**
+ * Returns true if the lock is too expensive for user in a given currency
+ * @param {*} lock
+ * @param {*} account
+ * @param {string} currencyKey
+ * @returns {boolean}
+ */
+export function isTooExpensiveForUserByCurrency(lock, account, currencyKey) {
+  // Not having an account or an account balance should be treated like having 0
+  // of whatever currency.
+  if (!account || !account.balance) {
+    return LOCK_IS_TOO_EXPENSIVE
+  }
+
+  const balance = parseFloat(account.balance[currencyKey])
+  if (!balance) {
+    return LOCK_IS_TOO_EXPENSIVE
+  }
+
+  const keyPrice = parseFloat(lock.keyPrice)
+  if (!keyPrice) {
+    // All locks should have a keyPrice, but if for some reason one doesn't,
+    // we're in some kind of broken state and we'd better prevent purchasing.
+    return LOCK_IS_TOO_EXPENSIVE
+  }
+
+  if (keyPrice <= balance) {
+    // This doesn't take gas into account. If keyPrice and balance are exactly
```

Comment:
So I think this is only true if this is an Ether lock. For ERC20 locks, we are probably fine with just the right amount?

Reply:
Let's to a strict equal for now and the crypto wallet will actually show an error on Ether locks when the user trying to purchase indicating that the balance is not high enough.

Reply:
If we do a strict `<` that would stop someone with exactly 20DAI from buying a key to a 20DAI lock, which feels like it might not be the right thing. With the current logic the wallet will still catch the balance being wrong once gas is accounted for.

Reply:
Oh sorry that was exactly what I wanted, I misread your code. We should not block for the strict equality!